emcc.py

```diff
@@ -1231,7 +1231,7 @@ def get_last_setting_change(setting):
 
     shared.Settings.TARGET_BASENAME = target_basename = unsuffixed_basename(target)
 
-    final_suffix = suffix(target)
+    final_suffix = get_file_suffix(target)
```

Comment:
This change is potentially more than a cleanup, I think?

Reply:
True, but I convinced myself this is safe (or even an improvement) :)

I checked all the places where final_suffix is used.. and all but one are checking its value against a known set of endings.

The only time is used in any other way it when no target output name is specified.

Reply:
Yeah, seems safe to me too.